projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-site-migration.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Allow us to access the Migrate Guru site migration key via API.
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ */
+class WPCOM_REST_API_V2_Endpoint_Site_Migration {
+	/**
+	 * Class constructor
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'site-migration';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register our routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_data' ),
+					'permission_callback' => array( $this, 'can_access' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Permission callback for the REST route.
+	 *
+	 * @return boolean
+	 */
+	public function can_access() {
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Returns the migration key.
+	 *
+	 * @return string
+	 */
+	private function get_migration_key() {
+		// Only fetch the key if we have the classes.
+		if ( class_exists( 'MGWPSettings' ) && class_exists( 'MGInfo' ) ) {
```

Comment:
At a high level, I have a concern with exposing this data in an always-on fashion when the underlying Migrate Guru plugin only exposes the value via direct UI access in wp-admin. Given that we only need this exposed for a short amount of time, I am wondering whether we should do something less always-on?

Reply:
discussion here p1709894323883899-slack-C0Q664T29